tokio/src/executor/task/core.rs

```diff
@@ -103,7 +106,7 @@ impl<T: Future> Core<T> {
         self.stage = Stage::Consumed
     }
 
-    pub(super) fn poll<S>(&mut self, header: &Header<S>) -> Poll<T::Output>
+    pub(super) fn poll<S>(&mut self, header: &Header) -> Poll<T::Output>
```

Comment:
This change worries me, almost to the point that it should be marked as `unsafe`. Now, this basically allows doing an arbitrary pointer cast in safe code.

Reply:
I don't think it is any more unsafe than it was before, but I can mark it as such.

Reply:
Do you have a change in mind?

Reply:
Before you had to have a `Header<S>` to go with the `poll`. Now, you no longer need to have a `Header<S>` to "prove" that you can use that `S` (before the proof was also arguably pretty weak). It's tricky, because I also agree with you argument from Discord that `unsafe fn` makes the _whole_ function body an unsafe block, which isn't great. I think I'd be okay with this being discussed in a doc block for the fn instead.